initial scaffolding for renv cheat sheet

kevinushey opened the pull request:
This gives some scaffolding for an `renv` cheat sheet, alongside some of the topics that I think are worth highlighting therein.

I included a "core concepts" section just because I think most new users of `renv` haven't thought much about where R packages actually get installed and loaded, but it might be too much?

kevinushey committed:
initial scaffolding for renv cheat sheet

kevinushey commented on html/renv.qmd:
```diff
@@ -0,0 +1,90 @@
+## Core Workflows
+
+- `renv::init()`: Initialize **renv** in a new or existing project.
+- `renv::install()`: Install R packages.
+- `renv::update()`: Update installed packages to their latest-available version.
```
Or perhaps these should be in a separate section -- I'm not totally sure?

mine-cetinkaya-rundel replied:
Added "`install()` targets packages you name or declared in the project DESCRIPTION file; `update()` targets all packages out of date compared to the source they were installed from" to clarify

shannonpileggi reviewed (commented):
thank you for getting this started, this is going to be a great resource!!

mine-cetinkaya-rundel commented on html/renv.qmd:
```diff
@@ -0,0 +1,90 @@
+## Configuring renv
+
+- `?renv::config`: User-specific settings. Tweak how you'd like `renv` to behave in different scenarios.
+- `?renv::settings`: Project-level settings. These settings are persisted as part of the project, and will be active for any other collaborators in the project.
```
Added these in various places throughout the cheatsheet

mine-cetinkaya-rundel commented on html/renv.qmd:
````diff
@@ -0,0 +1,90 @@
+- Discuss `renv::embed()` and `renv::use()`? (Ways to declare package dependency versions directly in .R / .Rmd / .qmd files)
+- Worth discussing that `renv` infers a package's source from its DESCRIPTION file? And so requires the installer of that package (renv, pak, remotes) to encode Remote dependency information at installation time.
+- Mention that more documentation + advanced topics are described in pkgdown website at https://rstudio.github.io/renv/index.html
+```
````
Added these throughout

mine-cetinkaya-rundel commented on html/renv.qmd:
```diff
@@ -0,0 +1,90 @@
+  - implicit => scan R files for dependencies; e.g. called to library(dplyr)
+  - explicit => only scan project DESCRIPTION file; user must keep up-to-date
+- Mention that `renv::install()` supports custom remotes, e.g. `renv::install("rstudio/dplyr")`?
+- Mention `options(renv.config.pak.enabled = TRUE)` for pak integration?
```
Did not add these to the cheatsheet for now

mine-cetinkaya-rundel committed:
First pass of addressing TODO items and review comments

mine-cetinkaya-rundel committed:
Add collaboration guidance and clarify renv limitations

mine-cetinkaya-rundel committed:
Add guidance to test code before snapshotting and clarify .gitignore handling

mine-cetinkaya-rundel committed:
Render renv

mine-cetinkaya-rundel committed:
Add Quarto ipynb cache files to gitignore

mine-cetinkaya-rundel committed:
Merged origin/main into kevinushey-renv

kevinushey committed:
incorporate review feedback

- move core concepts up; add the global package cache
- add status() triage, repair(), .renvignore, and a CI pointer
- fix rstudio/dplyr -> tidyverse/dplyr example
- register renv.qmd in the site index
- disable pdf preview link until the pdf version exists

kevinushey committed:
move status() triage into a staying in sync section

mine-cetinkaya-rundel committed:
Merged origin/main into kevinushey-renv

mine-cetinkaya-rundel committed:
Add key + pdf
